reference-implementation/lib/readable-stream.js

```diff
@@ -158,8 +158,70 @@ class ReadableStream {
     const branches = ReadableStreamTee(this, false);
     return createArrayFromList(branches);
   }
+
+  getIterator({ preventCancel = false } = {}) {
+    if (IsReadableStream(this) === false) {
+      throw streamBrandCheckException('getIterator');
+    }
+    const reader = AcquireReadableStreamDefaultReader(this);
+    const iterator = Object.create(ReadableStreamAsyncIteratorPrototype);
+    iterator._asyncIteratorReader = reader;
+    iterator._preventCancel = Boolean(preventCancel);
+    return iterator;
+  }
 }
 
+const AsyncIteratorPrototype = Object.getPrototypeOf(Object.getPrototypeOf(async function* () {}).prototype);
+const ReadableStreamAsyncIteratorPrototype = Object.setPrototypeOf({
```

Comment:
New tests, new bugs. This way of setting up the prototype chain causes `next()` and `return()` to become enumerable properties... 🙄 

I guess I'll just resort to `Object.defineProperty` again? Or perhaps `Object.create` with a property descriptor map.

Reply:
Hmmm... What's the cleanest way of creating a function with `.name === 'return'`? The following does not work, because `return` is a keyword:
```
Object.defineProperty(ReadableStreamAsyncIteratorPrototype, 'return', {
  value: function return(value) { ... },
  enumerable: false,
  writable: true,
  configurable: true
});
```

Reply:
It's simplest to just use object literals and then do `Object.defineProperty(RSAIP, 'return', { enumerable: false })`

Reply:
I am not proud of this contraption:
```js
const AsyncIteratorPrototype = Object.getPrototypeOf(Object.getPrototypeOf(async function* () {}).prototype);
const ReadableStreamAsyncIteratorMethods = class {
  next() { /* ... */ }
  return(value) { /* ... */}
};
const ReadableStreamAsyncIteratorPrototype = Object.create(AsyncIteratorPrototype, {
  next: Object.getOwnPropertyDescriptor(ReadableStreamAsyncIteratorMethods.prototype, 'next'),
  return: Object.getOwnPropertyDescriptor(ReadableStreamAsyncIteratorMethods.prototype, 'return')
});
```
I can't come up with any other way to have a function whose `name` equals `return` though... Please send help! 😆

Reply:
The following should work:

```js
const RSAIP = Object.setPrototypeOf({ next() { ... }, return() { ... } }, AIP);
Object.defineProperty(RSAIP, 'next', { enumerable: false });
Object.defineProperty(RSAIP, 'return', { enumerable: false });
```